hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java

```diff
@@ -674,4 +679,47 @@ private int countSerials(final List<NftAllowance> nftAllowancesList) {
         }
         return totalSerials;
     }
+
+    /**
+     * Counts the number of NEW allowances that will be added to state from the transaction.
+     * Compares transaction allowances against existing allowances in the account to determine
+     * which ones are new vs. updates to existing allowances.
+     *
+     * @param op the crypto approve allowance transaction body
+     * @param account the account existing in state (may be null if account doesn't exist)
+     * @return the count of new allowances that will be added to state
+     */
+    private long countNewAllowances(final CryptoApproveAllowanceTransactionBody op, final Account account) {
+        final long newCryptoKeys =
+                getChangedCryptoKeys(op.cryptoAllowances(), account == null ? emptyList() : account.cryptoAllowances());
+        final long newTokenKeys =
+                getChangedTokenKeys(op.tokenAllowances(), account == null ? emptyList() : account.tokenAllowances());
+        final long newApproveForAllNfts = getChangedNftKeys(
+                op.nftAllowances(), account == null ? emptyList() : account.approveForAllNftAllowances());
+
+        return newCryptoKeys + newTokenKeys + newApproveForAllNfts;
+    }
+
+    @NonNull
+    @Override
+    public FeeResult calculateFeeResult(@NonNull final FeeContext feeContext) {
+        requireNonNull(feeContext);
+        final var model = FeeModelRegistry.lookupModel(HederaFunctionality.CRYPTO_APPROVE_ALLOWANCE);
+        final var op = feeContext.body().cryptoApproveAllowanceOrThrow();
+
+        // Look up payer account to check existing allowances
+        final var accountStore = feeContext.readableStore(ReadableAccountStore.class);
+        final var account = accountStore.getAccountById(feeContext.payer());
+
+        final Map<Extra, Long> params = new HashMap<>();
+        params.put(Extra.SIGNATURES, (long) feeContext.numTxnSignatures());
+
+        // Count only NEW allowances that will be added to state
+        final long newAllowanceCount = countNewAllowances(op, account);
+        params.put(Extra.ALLOWANCES, newAllowanceCount);
```

Comment:
In one of the future PRs when we implement the interface Michael, proposed this will be cleaner. This can also be done in next PRs, just writing as I see it.